### ISSUE TITLE
[ED issue] Do we need Profile Resolution Semantics ?

### ISSUE THREAD

**Issue by cconcolato** [w3c/dapt#43]
Contrary to IMSC which has legacy reasons to use multiple ways to provide the profile, in DAPT, we require using `ttp:contentProfiles`. We should able to get rid of the "Profile Resolution" section.

**Comment by nigelmegitt** [comment]
Agreed, we should look at that section and either remove it or reduce it greatly.

**Comment by cconcolato** [comment]
We should make sure that it is acceptable for a document to also list other profile designators such as the one from IMSC if the document is compliant.

**Comment by nigelmegitt** [comment]
I've reviewed TTML2's section on [Profile Semantics](https://www.w3.org/TR/ttml2/#semantics-profile-semantics), and it looks to me that:

#### Content Profile

By requiring `ttp:contentProfiles` the [Construct Effective Content Profile](https://www.w3.org/TR/ttml2/#semantics-procedure-construct-effective-content-profile) procedure will either:
1. (step 1) Return with some application specific profile that's out of our scope here OR
2. (step 2) Include the requirements of DAPT as we want it to, alongside any other signalled content profiles.

#### Processor Profile

By not requiring `ttp:processorProfiles` we are leaving things somewhat open.

If a document does specify `ttp:processorProfiles` then it is saying what processor requirements are needed to process that document. That could be something related to DAPT or something else entirely.

**Question: Do we care? Presumably a non-DAPT processor would have its own rules and constraints, and we are only specifying the semantics and requirements of processors that are DAPT ones.**

If a document does not specify `ttp:processorProfiles` but does specify `ttp:profile` then step 4 in [Construct Effective Processor Profile](https://www.w3.org/TR/ttml2/#semantics-procedure-construct-effective-processor-profile) comes into action. If the value in `ttp:profile` differs from or is incompatible with `ttp:contentProfiles` then that's a weird thing.

**Proposal Permit `ttp:profile` but constrain it not to be incompatible with DAPT.**

If a document specifies neither `ttp:processorProfiles` nor `ttp:profile` then step 5 in [Construct Effective Processor Profile](https://www.w3.org/TR/ttml2/#semantics-procedure-construct-effective-processor-profile) comes into action, which in this case results in the profile combination specified by `ttp:contentProfiles`, which is fine.

**Comment by cconcolato** [comment]
Regarding `ttp:profile`, I would prefer not permitting it or recommending not setting it. It makes things simpler. But if the group feels that the proposal to permit it but only with values not incompatible with DAPT is fine, I'm good with that as long as we make it clear for authors. I don't want authors to start putting values and not realizing they are incompatible with DAPT.
I would be happy with not permitting or recommending not to set `ttp:processorProfiles` at all.

**Comment by nigelmegitt** [comment]
Editorial chat with @cconcolato and me 2022-12-15:

Outcome we want:
* The document clearly signals that it is a DAPT document
* A processor that supports DAPT must not infer that it can't, from `processorProfiles`
* Future versions of the spec may add features - a processor that supports them needs to know it can process the document, but a processor that does not support them should not attempt to process the document? Or should try but ignore what it doesn't know? We should only add backwards compatibility non-breaking features.
* Processors that do not understand vocabulary defined in later versions of the spec should ignore it but continue with v1 functionality.
* Therefore we don't want to change the profile designator when we introduce non-breaking changes.
* But we (or others) may add extra profile designators that can be used in `processorProfiles` so that document authors or tools can effectively require processors that understand the new features.
* We might need to add for any future version features the profile designator that applies to them.
* We need a para to explain the 3 mechanisms in TTML2 and say:
  * `ttp:profile` is not permitted
  * `ttp:processorProfiles` may be present, with informative note explaining the purpose and that in future versions we may introduce features associated with new profile designators, and if that is (additionally) present in `contentProfiles` it is informative, but if it is (additionally) present in `processorProfiles` then it cannot be ignored and processors that do not support the features associated with that new designator should not process the document.
  * `ttp:contentProfiles` must be present, and must indicate all the DAPT profiles whose features are used in the document.

**Comment by nigelmegitt** [comment]
I've been wondering what to do about all the TTML2 [Profile Attribute Vocabulary](https://www.w3.org/TR/ttml2/#profile-attribute-vocabulary) that may also be present, and how to allow people to use it if they want, while not requiring it in general.

It could be a major distraction to implementers if they are required to process profiles to decide if they can process a document.

On the other hand, what if some document author has a particular need for a feature, and only wants a processor to handle the document if it can support that feature?

I think the best way through this is effectively to require any processor profile combination to be done externally to the document instance, so we would require support for [`#contentProfiles`](https://www.w3.org/TR/ttml2/#feature-contentProfiles), and permit (optional) support for [`#processorProfiles`](https://www.w3.org/TR/ttml2/#feature-processorProfiles) but not require any support for [`#contentProfiles-combined`](https://www.w3.org/TR/ttml2/#feature-contentProfiles-combined) or [`#processorProfiles-combined`](https://www.w3.org/TR/ttml2/#feature-processorProfiles-combined).

I thought about defining a profile whose required features are _only_ the additional profile processing-related ones we could require, i.e. those from [`#profile-full-version-2`](https://www.w3.org/TR/ttml2/#feature-profile-full-version-2) (but excluding `#profile`), which are:

* [#contentProfiles-combined](https://www.w3.org/TR/ttml2/#feature-contentProfiles-combined)
* [#permitFeatureNarrowing](https://www.w3.org/TR/ttml2/#feature-permitFeatureNarrowing)
* [#permitFeatureWidening](https://www.w3.org/TR/ttml2/#feature-permitFeatureWidening)
* [#processorProfiles](https://www.w3.org/TR/ttml2/#feature-processorProfiles)
* [#processorProfiles-combined](https://www.w3.org/TR/ttml2/#feature-processorProfiles-combined)
* [#profile-nesting](https://www.w3.org/TR/ttml2/#feature-profile-nesting)
* [#profile-version-2](https://www.w3.org/TR/ttml2/#feature-profile-version-2)

And then give that profile a well known URL (we'd publish it on /TR or similar) and, for any document whose processing somehow depends on it, then we would require that URL is the first one in the list in `tt@ttp:processorProfiles`.

But on reflection, that (second approach) would be solving a problem we don't yet have, so I think we can leave that to another day, in case we ever need it.

**Comment by nigelmegitt** [comment]
PR #103 open - the one thing I did not do in that PR was write any text about the intent for future- and back- compatibility.
